tests/runtests.py

```diff
@@ -1,15 +1,60 @@
-#!/usr/bin/env python
-# Copyright 2015-2021 XMOS LIMITED.
+# Copyright 2015-2024 XMOS LIMITED.
 # This Software is subject to the terms of the XMOS Public Licence: Version 1.
-import xmostest
+from pathlib import Path
+import subprocess
+import pytest
+import re
+
+test_path = Path(__file__).parent
+
+@pytest.mark.parametrize("test_name", [b.stem for b in test_path.rglob("*.xe")])
+def test(test_name):
+   test_bin = [b for b in test_path.rglob(f"{test_name}.xe")][0]
+   expect_file = test_path / f"{test_name}.expect"
+   with open(expect_file, "r") as fp:
+      expected_output = fp.read()
+      cmd = ["xsim", test_bin]
```

Comment:
@danielpieczko should comment on whether this simple case (no plugins etc) should be using his new test_support/pyxsim rather than manual call to xsim

Reply:
I think direct xsim calls are fine for situations like this where the test code is very simple; we can save pyxsim for the more complex situation.